pygmt/src/meca.py

```diff
@@ -0,0 +1,413 @@
+"""
+meca - Plot focal mechanisms.
+"""
+
+import numpy as np
+import pandas as pd
+from pygmt.clib import Session
+from pygmt.exceptions import GMTError, GMTInvalidInput
+from pygmt.helpers import (
+    build_arg_string,
+    data_kind,
+    dummy_context,
+    fmt_docstring,
+    kwargs_to_strings,
+    use_alias,
+)
+
+
+@fmt_docstring
+@use_alias(
+    R="region",
+    J="projection",
+    B="frame",
+    C="offset",
+    N="no_clip",
+    V="verbose",
+    X="xshift",
+    Y="yshift",
+    p="perspective",
+    t="transparency",
+)
+@kwargs_to_strings(R="sequence", p="sequence")
+def meca(
+    spec,
+    scale,
+    longitude=None,
+    latitude=None,
+    depth=None,
+    convention=None,
+    component="full",
+    plot_longitude=None,
+    plot_latitude=None,
+    **kwargs,
+):
+    """
+    Plot focal mechanisms.
+
+    Full option list at :gmt-docs:`supplements/seis/meca.html`
+
+    Note
+    ----
+        Currently, labeling of beachballs with text strings is only
+        supported via providing a file to `spec` as input.
```

Comment:
The module has been taken out of the Python `class` which frees up 4 spaces, so should I re-wrap the lines to fit the 79 character limit here?

Reply:
I think yes.